gazell improvement (#13682)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

|no_changes_yet_note|

Gaezll
------

* Added:

   * User-specific file for Gazell pairing. User can override Gazell pair config inside application.

>    * Clear system address and host id in RAM when ``gzp_erase_pairing_data`` is called.

Reviewer: The clearing of RAM variables may be considered fixes.

|no_changes_yet_note|

> Gaezll

Reviewer: Typo. It should be Gazell.

------

* Added:

>    * User-specific file for Gazell pairing. User can override Gazell pair config inside application.

Reviewer: This changelog line is added in another commit.

Author: Separate changelog to 2 commits.

>    * User-specific file for Gazell pairing. User can override Gazell pair config inside application.

Reviewer 2: ```suggestion
   * User-specific file for Gazell pairing.
     User can override Gazell pair config inside the application.
```

Reviewer 3: Consider referring to Kconfig option names to simplify looking for newly added options e.g.:

```suggestion
   * :kconfig:option:`CONFIG_GAZELL_PAIRING_CONFIG_USER` and :kconfig:option:`CONFIG_GAZELL_PAIRING_CONFIG_FILE_NAME`.
      The options allow to use user-specific file as Gazell pairing configuration header to override the pairing configuration.
```

* Fixed:

>    * Clear system address and host id in RAM when ``gzp_erase_pairing_data`` is called.

Reviewer: The C function style should make it :c:func:`gzp_erase_pairing_data`.

Reviewer: The C function style should be :c:func:\`gzp_erase_pairing_data\`. Please search the releases notes for examples.

Reviewer: Technical writer has the same comment. Resolve this one.

>    * Clear system address and host id in RAM when :c:func:gzp_erase_pairing_data is called.

Reviewer 2: ```suggestion
   * Clear system address and host id in RAM when :c:func:`gzp_erase_pairing_data` is called.
```
